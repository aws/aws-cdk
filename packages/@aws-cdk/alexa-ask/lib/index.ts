// Alexa::ASK CloudFormation Resources:
export * from './ask.generated';
export * from './pipeline-actions';
