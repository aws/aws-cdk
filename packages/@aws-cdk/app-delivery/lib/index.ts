export * from './pipeline-deploy-stack-action';
