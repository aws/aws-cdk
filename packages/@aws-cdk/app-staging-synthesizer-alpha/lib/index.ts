export * from './default-staging-stack';
export * from './app-staging-synthesizer';
export * from './bootstrap-roles';
export * from './staging-stack';
