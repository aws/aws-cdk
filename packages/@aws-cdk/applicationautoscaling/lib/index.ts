// AWS::ApplicationAutoScaling CloudFormation Resources:
export * from './applicationautoscaling.generated';
