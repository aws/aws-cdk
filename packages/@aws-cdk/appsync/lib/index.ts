// AWS::AppSync CloudFormation Resources:
export * from './appsync.generated';
