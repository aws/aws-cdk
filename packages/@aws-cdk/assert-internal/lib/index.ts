export * from './assertion';
export * from './canonicalize-assets';
export * from './expect';
export * from './inspector';
export * from './synth-utils';

export * from './assertions/exist';
export * from './assertions/have-output';
export * from './assertions/have-resource';
export * from './assertions/have-resource-matchers';
export * from './assertions/have-type';
export * from './assertions/match-template';
export * from './assertions/and-assertion';
export * from './assertions/negated-assertion';
export * from './assertions/count-resources';
