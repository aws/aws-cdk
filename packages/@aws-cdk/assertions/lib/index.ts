export * from './assertions';
export * from './match';