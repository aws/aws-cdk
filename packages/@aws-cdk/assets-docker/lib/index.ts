// tslint:disable-next-line:no-console
console.error('the @aws-cdk/assets-docker is deprecated. use @aws-cdk/aws-ecr-assets');
