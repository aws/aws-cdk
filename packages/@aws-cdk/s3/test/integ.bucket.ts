#!/usr/bin/env node
import { App, Stack } from '@aws-cdk/core';
import { User } from '@aws-cdk/iam';
import { Bucket, BucketEncryption } from '../lib';

const app = new App(process.argv);

const stack = new Stack(app, 'aws-cdk-s3');

const bucket = new Bucket(stack, 'MyBucket', {
    encryption: BucketEncryption.Kms
});

const user = new User(stack, 'MyUser');
bucket.grantReadWrite(user);

process.stdout.write(app.run());
