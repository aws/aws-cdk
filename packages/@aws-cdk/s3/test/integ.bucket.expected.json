{
  "Resources": {
    "MyBucketKeyC17130CF": {
      "Type": "AWS::KMS::Key",
      "Properties": {
        "Description": "Created by aws-cdk-s3/MyBucket",
        "KeyPolicy": {
          "Statement": [
            {
              "Action": [
                "kms:Create*",
                "kms:Describe*",
                "kms:Enable*",
                "kms:List*",
                "kms:Put*",
                "kms:Update*",
                "kms:Revoke*",
                "kms:Disable*",
                "kms:Get*",
                "kms:Delete*",
                "kms:ScheduleKeyDeletion",
                "kms:CancelKeyDeletion"
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition"
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId"
                      },
                      ":root"
                    ]
                  ]
                }
              },
              "Resource": "*"
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*"
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "MyUserDC45028B",
                    "Arn"
                  ]
                }
              },
              "Resource": "*"
            }
          ],
          "Version": "2012-10-17"
        }
      },
      "DeletionPolicy": "Retain"
    },
    "MyBucketF68F3FF0": {
      "Type": "AWS::S3::Bucket",
      "Properties": {
        "BucketEncryption": {
          "ServerSideEncryptionConfiguration": [
            {
              "ServerSideEncryptionByDefault": {
                "KMSMasterKeyID": {
                  "Fn::GetAtt": [
                    "MyBucketKeyC17130CF",
                    "Arn"
                  ]
                },
                "SSEAlgorithm": "aws:kms"
              }
            }
          ]
        }
      }
    },
    "MyBucketPolicyE7FBAC7B": {
      "Type": "AWS::S3::BucketPolicy",
      "Properties": {
        "Bucket": {
          "Ref": "MyBucketF68F3FF0"
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:PutObject*",
                "s3:DeleteObject*",
                "s3:Abort*"
              ],
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::GetAtt": [
                    "MyUserDC45028B",
                    "Arn"
                  ]
                }
              },
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "MyBucketF68F3FF0",
                    "Arn"
                  ]
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "MyBucketF68F3FF0",
                          "Arn"
                        ]
                      },
                      "/",
                      "*"
                    ]
                  ]
                }
              ]
            }
          ],
          "Version": "2012-10-17"
        }
      }
    },
    "MyUserDC45028B": {
      "Type": "AWS::IAM::User"
    },
    "MyUserDefaultPolicy7B897426": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:PutObject*",
                "s3:DeleteObject*",
                "s3:Abort*"
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::GetAtt": [
                    "MyBucketF68F3FF0",
                    "Arn"
                  ]
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      {
                        "Fn::GetAtt": [
                          "MyBucketF68F3FF0",
                          "Arn"
                        ]
                      },
                      "/",
                      "*"
                    ]
                  ]
                }
              ]
            },
            {
              "Action": [
                "kms:Decrypt",
                "kms:DescribeKey",
                "kms:Encrypt",
                "kms:ReEncrypt*",
                "kms:GenerateDataKey*"
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "MyBucketKeyC17130CF",
                  "Arn"
                ]
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "PolicyName": "MyUserDefaultPolicy7B897426",
        "Users": [
          {
            "Ref": "MyUserDC45028B"
          }
        ]
      }
    }
  }
}